src/main/java/org/apache/parquet/hadoop/BdecParquetWriter.java

```diff
@@ -264,7 +265,7 @@ public void prepareForWrite(RecordConsumer recordConsumer) {
 
     @Override
     public void write(List<Object> values) {
-      List<ColumnDescriptor> cols = schema.getColumns();
+      List<Type> cols = schema.getFields();
```

Comment:
what's the difference between getColumns and getFields?

Reply:
`getFields ` does not flatten the subcolumns. This change is going to be used in structured data, reverted here and will be in next PR.